src/main/java/net/dv8tion/jda/webhook/WebhookClient.java

```diff
@@ -65,19 +66,23 @@
     protected final OkHttpClient client;
     protected final ScheduledExecutorService pool;
     protected final Bucket bucket;
-    protected final BlockingQueue<Pair<RequestBody, CompletableFuture<?>>> queue;
+    protected final BlockingQueue<Pair<RequestBody, CompletableFuture<SentWebhookMessage>>> queue;
     protected volatile boolean isQueued;
     protected boolean isShutdown;
+    protected boolean isWait;
 
-    protected WebhookClient(final long id, final String token, final OkHttpClient client, final ScheduledExecutorService pool)
+    private final EntityBuilder entityBuilder = new EntityBuilder(null);
```

Comment:
This is even worse, why not just make a static method?

Reply:
Because that would require many methods to become static as well. One of them being EntityBuilder#getJDA

Reply:
No that is not true, your factories will not need a JDA instance and thus don't need state